crates/re_space_view_bar_chart/src/space_view_class.rs

```diff
@@ -106,54 +110,15 @@ impl SpaceViewClass for BarChartSpaceView {
 
     fn selection_ui(
         &self,
-        _ctx: &ViewerContext<'_>,
+        ctx: &ViewerContext<'_>,
         ui: &mut egui::Ui,
-        _state: &mut dyn SpaceViewState,
+        state: &mut dyn SpaceViewState,
         _space_origin: &EntityPath,
-        _space_view_id: SpaceViewId,
-        root_entity_properties: &mut EntityProperties,
+        space_view_id: SpaceViewId,
+        _root_entity_properties: &mut EntityProperties,
     ) -> Result<(), SpaceViewSystemExecutionError> {
-        ui.selection_grid("bar_chart_selection_ui").show(ui, |ui| {
-            ui.grid_left_hand_label("Legend");
-
-            ui.vertical(|ui| {
-                let mut selected = *root_entity_properties.show_legend.get();
-                if ui.re_checkbox(&mut selected, "Visible").changed() {
-                    root_entity_properties.show_legend = EditableAutoValue::UserEdited(selected);
-                }
-
-                let mut corner = root_entity_properties
-                    .legend_location
-                    .unwrap_or(LegendCorner::RightTop);
-
-                egui::ComboBox::from_id_source("legend_corner")
-                    .selected_text(corner.to_string())
-                    .show_ui(ui, |ui| {
-                        ui.selectable_value(
-                            &mut corner,
-                            LegendCorner::LeftTop,
-                            LegendCorner::LeftTop.to_string(),
-                        );
-                        ui.selectable_value(
-                            &mut corner,
-                            LegendCorner::RightTop,
-                            LegendCorner::RightTop.to_string(),
-                        );
-                        ui.selectable_value(
-                            &mut corner,
-                            LegendCorner::LeftBottom,
-                            LegendCorner::LeftBottom.to_string(),
-                        );
-                        ui.selectable_value(
-                            &mut corner,
-                            LegendCorner::RightBottom,
-                            LegendCorner::RightBottom.to_string(),
-                        );
-                    });
-
-                root_entity_properties.legend_location = Some(corner);
-            });
-            ui.end_row();
+        list_item::list_item_scope(ui, "time_series_selection_ui", |ui| {
```

Comment:
Not necessarily this PR, but we should probably put that scope just outside of the `selection_ui` call in `selection_panel.rs`, and remove them all from the `selection_ui` impls.